### PR TITLE
Miscellaneous cleanup for old issues.

### PR DESCRIPTION
**Pull request by ghost**
As the title says, just cleaning up some old stuff.

### CODE REVIEW

File: src/librustc/lint/mod.rs [hunk 1]
```diff
@@ -89,7 +89,6 @@ macro_rules! lint_initializer {
 /// Declare a static item of type `&'static Lint`.
 #[macro_export]
 macro_rules! declare_lint {
-    // FIXME(#14660): deduplicate
```
Comment (bluss) [inline src/librustc/lint/mod.rs]: I think this one can stay? Intention seems to be to reduce the duplication in the two cases.

**Comment by bluss** [comment]
The rest looks good to me! Nice work to clean up all those small old items.

**Comment by bluss** [comment]
a reviewer can r=me with the one inline comment addressed.

File: src/librustc/lint/mod.rs (hunk 1 above)
Comment (Eljay) [inline src/librustc/lint/mod.rs]: I removed it because it seems impossible to do so now, you can't place a `pub` qualifier on a macro invocation (#22285).

Comment (bluss) [inline src/librustc/lint/mod.rs]: Ok, yep, the FIXME is obsoleted now.

**Comment by bluss** [comment]
@bors r+

thank you!

**Pull request merged**